macros/src/main/scala/com/evolutiongaming/kryo/Serializer.scala

```diff
@@ -176,6 +181,63 @@ object Serializer {
             val lf = genReader(lt.dealias)
             val rf = genReader(rt.dealias)
             q"if (input.readInt == 0) scala.util.Left($lf) else scala.util.Right($rf)"
+          } else if (tpe <:< typeOf[mutable.LongMap[_]]) {
+            val t = tpe.typeArgs.head.dealias
+            val f = genReader(t)
+            val comp = companion(tpe)
+            q"""
+               val size = input.readInt
+               val map = $comp.empty[$t]
+               var i = 0
+               while (i < size) {
+                 map.update(input.readLong, $f)
+                 i += 1
+               }
+               map
+             """
```

Comment:
Maybe it makes sense to generate a reader/writer function for each Map/Set type instead of inlining the code. Imagine a lot of classes with lots of Map fields – this will blow up the size of serializer methods.

Reply:
@nau please see https://github.com/evolution-gaming/kryo-macros/pull/7/files